websites/K/Kick/presence.ts

```diff
@@ -0,0 +1,204 @@
+	smallImageKey?: string;
+	largeImageKey?: string;
+	state?: string;
+	buttons?: [ButtonData, ButtonData?];
+} {
+	switch (hostname) {
+		case "kick.com": {
+			switch (page) {
+				case "":
+					return { details: "Viewing home...", smallImageKey: Assets.Search };
+				case "categories": {
+					const activeMainCategory = document.querySelector(
+						"a.category-tile-active"
+					)?.textContent;
+					let state = "",
+						largeImageKey = "";
+
+					if (activeMainCategory) {
+						state = activeMainCategory;
+						// Replace .gif with .png on main category image
+						const imageElement = document.querySelector(
+								'a.category-tile-active [src$=".gif"]'
+							),
+							newSrc = imageElement.getAttribute("src").replace(".gif", ".png");
+						imageElement.setAttribute("src", newSrc);
+						largeImageKey = newSrc;
+					}
+					// If there is a subcategory
+					if (pageDetail && title) {
+						state += ` > ${formatText(pageDetail)} > ${formatText(title)}`;
+						largeImageKey =
+							document.querySelector<HTMLImageElement>("div.h-full > img")?.src;
+					}
+					return {
+						details: "Viewing categories...",
+						state,
+						largeImageKey,
+						smallImageKey: Assets.Search,
+					};
+				}
+				case "following": {
+					return {
+						details: "Viewing following...",
+						state: `${
+							document.querySelector(".\\!border-primary\\/100").textContent
+						} section`,
+						smallImageKey: Assets.Search,
+					};
+				}
+				case "dashboard": {
+					let state = document.querySelector(
+						".router-link-active .item-title"
+					).textContent;
+					const isContentExpanded = document.querySelector(
+						".content-expanded .item-title"
+					);
+
+					if (isContentExpanded) {
+						state = `${isContentExpanded.textContent} >
+							${
+								document.querySelector(
+									".content-expanded .router-link-active .item-title"
+								).textContent
+							}`;
+					}
+					return {
+						details: `Viewing ${formatText(page)}...`,
+						state,
+						smallImageKey: Assets.Viewing,
+					};
+				}
+				case "transactions": {
+					return {
+						details: `Viewing ${formatText(page)}...`,
+						state: formatText(pageDetail),
+					};
+				}
+				case "community-guidelines":
+				case "dmca-policy":
+				case "privacy-policy":
+				case "terms-of-service":
+					return {
+						details: `Reading ${formatText(page)}...`,
+						smallImageKey: Assets.Reading,
+					};
+				default:
+					// watching/viewing a stream
+					if (document.querySelector(".stream-username")) {
+						let smallImageKey = "",
+							state = "",
+							buttons: [ButtonData, ButtonData?];
+						const streamer =
+							document.querySelector(".stream-username").textContent;
+						if (document.querySelector(".odometer-value")) {
+							state = `Watching: ${streamer}`;
+							smallImageKey = Assets.Live;
+							buttons = [
+								{
+									label: "Watch stream",
+									url: `https://kick.com/${streamer}`,
```

Comment:
Couldn't `href` do the job here?

Reply:
do you mean like  document.location.href?

Reply:
> do you mean like document.location.href?

yes